ios/MullvadVPN/Operations/AsyncOperation.swift

```diff
@@ -147,3 +353,19 @@ extension Operation {
         }
     }
 }
+
+extension Operation {
+    var operationName: String {
+        return name ?? "\(self)"
+    }
+}
+
+
+protocol OperationBlockObserverSupport {}
+extension AsyncOperation: OperationBlockObserverSupport {}
+
+extension OperationBlockObserverSupport where Self: AsyncOperation {
+    func addBlockObserver(_ observer: OperationBlockObserver<Self>) {
+        addObserver(observer)
+    }
+}
```

Comment:
The reason for that twist is because we cannot use `Self` in class definition.

The following code does not compile when it's a part of class definition:

```
func addBlockObserver(_ observer: OperationBlockObserver<Self>) {
  // ...
}
```

The compiler would insist that we need to change `OperationBlockObserver<Self>` to `OperationBlockObserver <AsyncOperation>` which is not useful for subclasses.

However we can reference `Self` in protocol extensions and compiler properly infers `Self` type in that case.